src/matrix/room/members/load.js

```diff
@@ -90,3 +90,9 @@ export async function fetchOrLoadMembers(options, logger) {
         return loadMembers(options);
     }
 }
+
+export async function loadMember({roomId, userId, storage}) {
```

Comment:
We need a `fetchOrLoadMember` so that we can  show the details of a member who has not yet been fetched (roomSummary.hasFetchedMembers = false; since memberlist has not been loaded yet).  
We could do this using the [/profile](https://matrix.org/docs/spec/client_server/r0.6.1#get-matrix-client-r0-profile-userid) endpoint.

This is only a problem when switching to another room which has not fetched the memberlist yet but does contain the member.

Reply:
Yeah, I think the best approach here would be:

1. try read the member
2. if found, return
2. if not found, fetch it with https://matrix.org/docs/spec/client_server/latest#get-matrix-client-r0-rooms-roomid-state-eventtype-statekey
3. store the member
4. return

The idea being that if you've opened the member panel before, it will be available offline the next time you open it.